components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/password/NotificationPasswordRecoveryManager.java

```diff
@@ -670,7 +690,15 @@ public User updateUserPassword(String code, String confirmationCode, String pass
             validateTenantDomain(userRecoveryData.getUser());
             int failedAttempts = userRecoveryData.getFailedAttempts();
 
-            if (!StringUtils.equals(code, userRecoveryData.getSecret())) {
+            String hashedCode;
+            try {
+                hashedCode = Utils.doHash(code);
+            } catch (UserStoreException e) {
+                throw Utils.handleServerException(
+                        IdentityRecoveryConstants.ErrorMessages.ERROR_CODE_NO_HASHING_ALGO_FOR_CODE, null);
+            }
+            if (!StringUtils.equals(hashedCode, userRecoveryData.getSecret()) && !StringUtils.equals(code,
+                    userRecoveryData.getSecret())) {
```

Comment:
```suggestion
            if (!(StringUtils.equals(hashedCode, userRecoveryData.getSecret()) || StringUtils.equals(code,
                    userRecoveryData.getSecret()))) {
```

This will improve the readability.

Reply:
Fixed with c5821975215ed0a9ab572c698e20b59881a214e9